### PR TITLE
KOGITO-4921 - Add artifactId and version to generated dashboard title and name

### PR DESCRIPTION
**Pull request by r00ta**
JIRA: https://issues.redhat.com/browse/KOGITO-4921

Resulting grafana page is the following

Screenshot from 2021-04-14 10-12-32

Kogito-example pr: https://github.com/kiegroup/kogito-examples/pull/656

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-ide-config)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by r00ta** [comment]
Jenkins retest this please

**Comment by r00ta** [comment]
@robpblake fyi

File: kogito-codegen-modules/kogito-codegen-decisions/src/main/java/org/kie/kogito/codegen/decision/DecisionCodegen.java [hunk 1]
```diff
@@ -231,17 +231,21 @@ private void generateAndStoreGrafanaDashboards(DecisionRestResourceGenerator res
...
                 context().getAddonsConfig().useTracing());
         String domainDashboard = GrafanaConfigurationWriter.generateDomainSpecificDMNDashboard(
                 domainDashboardDmnTemplate,
+                dashboardName,
                 resourceGenerator.getNameURL(),
                 decisions,
                 context().getAddonsConfig().useTracing());
-        generatedFiles.addAll(DashboardGeneratedFileUtils.operational(operationalDashboard, resourceGenerator.getNameURL() + ".json"));
-        generatedFiles.addAll(DashboardGeneratedFileUtils.domain(domainDashboard, resourceGenerator.getNameURL() + ".json"));
+        generatedFiles.addAll(DashboardGeneratedFileUtils.operational(operationalDashboard, dashboardName + ".json"));
```
Comment (jiripetrlik) [inline kogito-codegen-modules/kogito-codegen-decisions/src/main/java/org/kie/kogito/codegen/decision/DecisionCodegen.java:247]: Would it be possible to add also a unit test to check these names are generated correctly?

File: kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java [hunk 2]
```diff
@@ -357,11 +359,13 @@ private void initRuleUnitHelper(RuleUnitHelper ruleUnitHelper, RuleUnitDescripti
...
         if (context().hasREST()) {
             if (context().getAddonsConfig().usePrometheusMonitoring()) {
+                String dashboardName = GrafanaConfigurationWriter.buildDashboardName(context().getGAV(), query.getEndpointName());
                 String dashboard = GrafanaConfigurationWriter.generateOperationalDashboard(
                         operationalDashboardDrlTemplate,
+                        dashboardName,
                         query.getEndpointName(),
                         context().getAddonsConfig().useTracing());
-                generatedFiles.addAll(DashboardGeneratedFileUtils.operational(dashboard, query.getEndpointName() + ".json"));
+                generatedFiles.addAll(DashboardGeneratedFileUtils.operational(dashboard, dashboardName + ".json"));
```
Comment (jiripetrlik) [inline kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java:368]: Would it be possible to add also a unit test to check these names are generated correctly?

**Review by jiripetrlik — COMMENTED** [review]
Hello @r00ta thank you for your PR. Maybe can you please explain some motivation for adding GAV to dashboard name? The name seems quite long now. But I don't have any strong opinion on this.

**Comment by r00ta** [comment]
> Hello @r00ta thank you for your PR. Maybe can you please explain some motivation for adding GAV to dashboard name? The name seems quite long now. But I don't have any strong opinion on this.

Hi @jiripetrlik , see the epic of the jira (https://issues.redhat.com/browse/BAAAS-138). In the scenario where different version of the application are deployed at the same time the SRE needs to know the mapping between dashboards and application version

File: kogito-codegen-modules/kogito-codegen-decisions/src/main/java/org/kie/kogito/codegen/decision/DecisionCodegen.java (hunk 1 above)
Comment (r00ta) [reply kogito-codegen-modules/kogito-codegen-decisions/src/main/java/org/kie/kogito/codegen/decision/DecisionCodegen.java:247]: Hi @jiripetrlik , we have such test in the kogito-examples projects https://github.com/kiegroup/kogito-examples/pull/656 . Is it ok or do you want to have unit tests also here?

Comment (jiripetrlik) [reply kogito-codegen-modules/kogito-codegen-decisions/src/main/java/org/kie/kogito/codegen/decision/DecisionCodegen.java:247]: @r00ta 
It depends how complicated this test will be. If it is only some kind of simple unit test which only checks that strings are concatenated in the right order etc. and check that the result is correct I think it is worth to have it also in the same repo as functionality is. If it is something more complex then I think the test in examples is enough. Will leave it on your decision.

**Review by jiripetrlik — APPROVED** [review]
@r00ta
Thank you for your answer. LGTM

**Comment by r00ta** [comment]
Jenkins retest this please

**Comment by r00ta** [comment]
Jenkins retest this please

File: kogito-codegen-modules/kogito-codegen-decisions/src/main/java/org/kie/kogito/codegen/decision/DecisionCodegen.java (hunk 1 above)
Comment (r00ta) [reply kogito-codegen-modules/kogito-codegen-decisions/src/main/java/org/kie/kogito/codegen/decision/DecisionCodegen.java:247]: Hi @jiripetrlik , I've added a unit test for the `GrafanaConfigurationWriter`. Let me know if this looks good to you.

**Comment by r00ta** [comment]
Jenkins retest this please

File: kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java (hunk 2 above)
Comment (r00ta) [reply kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java:368]: Same as above: added a unit test in `GrafanaConfigurationWriter`

**Review by danielezonca — APPROVED** [review]

**Comment by r00ta** [comment]
Jenkins retest this please

**Review by jiripetrlik — APPROVED** [review]
@r00ta
Thank you for the changes! LGTM

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
@r00ta 
`06:13:22,667 [ERROR] Failed to execute goal net.revelc.code.formatter:formatter-maven-plugin:2.13.0:validate (default) on project grafana-api: File '/home/jenkins/workspace/KIE/kogito/pullrequest/kogito-runtimes/kogito-runtimes/grafana-api/src/test/java/org/kie/kogito/grafana/GrafanaConfigurationWriterTest.java' has not been previously formatted.  Please format file and commit before running validation! -> [Help 1]`

Please fix formatting :)

**Comment by r00ta** [comment]
Jenkins retest this please

**Comment by r00ta** [comment]
Jenkins retest this please

**Pull request merged**